### PR TITLE
Fixed memory leak in scheduler HUP

### PR DESCRIPTION
**Pull request by bhroam**
#### Describe Bug or Feature
Memory leak when the scheduler is HUP'd

#### Describe Your Change
The scheduler dups some strings when it parsing server_dyn_res.  It does not free these strings on HUP.

#### Attach Test and Valgrind Logs/Output
BEFORE:
[bmann@mars scheduler]$ grep parse_config valgrind-before.log 
==814486==    by 0x44A932: parse_config (parse.c:611)
==814486==    by 0x44A932: parse_config (parse.c:611)
==814486==    by 0x44A9AA: parse_config (parse.c:620)
==814486==    by 0x44A9BA: parse_config (parse.c:621)
==814486==    by 0x44A9AA: parse_config (parse.c:620)
==814486==    by 0x44A9BA: parse_config (parse.c:621)

AFTER:
[bmann@mars scheduler]$ sudo valgrind --leak-check=full --track-origins=yes --log-file=valgrind.log ./pbs_sched -N
[bmann@mars scheduler]$ grep parse_config valgrind.log

### CODE REVIEW

File: src/scheduler/parse.c [hunk 1]
```diff
@@ -934,7 +943,6 @@ init_config()
 	conf.update_comments = 1;
 
 	conf.decay_time = 86400;	/* default decay time period is 24 hours */
-	conf.sync_time = 86400;
```
Comment (agrawalravi90) [inline src/scheduler/parse.c:937]: was this intentional?

Comment (bhroam) [reply src/scheduler/parse.c:937]: Oh, yeah, it was.  I can back it out if you want.  That's literally the only place sync time is set/used.

Comment (agrawalravi90) [reply src/scheduler/parse.c:937]: oh, so should we just remove conf.sync_time altogether?

Comment (bhroam) [reply src/scheduler/parse.c:937]: Sure, let me remove it.  I should probably file a doc bug to get it removed from the man page because you can't set it.

**Pull request merged**